hw/misc/hermes.c

```diff
@@ -819,15 +834,22 @@ static uint64_t hermes_bar2_msix_pba_write(hwaddr addr)
 
 static uint64_t hermes_bar2_read(void *opaque, hwaddr addr, unsigned size)
 {
+    int ch = addr2ch(addr);
     HermesState *hermes = opaque;
     uint64_t val = ~0ULL;
 
+    if (ch >= H2C_CHANNELS || ch >= C2H_CHANNELS) {
```

Comment:
Although unlikely, it is possible to have `H2C_CHANNELS != C2H_CHANNELS`, so we have to check against only one of them. Probably easier to do it in the switch below. Also, it is not exactly an error, more like just an invalid BAR address. I'd just use `hermes_bar_warn_invalid()`.